### PR TITLE
Generate all valid parentheses - Java

### PR DESCRIPTION
**Pull request by shahina-bano**
Added the code for issue #732 in Java. kindly review it @shreyanspoddar @raksha009

### CODE REVIEW

File: Code/Java/valid_parantheses.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*Generate all valid parantheses
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:1]:
```suggestion
/*
Generate all valid parentheses

```

File: Code/Java/valid_parantheses.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*Generate all valid parantheses
+Problem  Statement: Write a program to generate all possible "n" pairs of balanced parantheses.
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:2]: Describe in a bit more detail.

File: Code/Java/valid_parantheses.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import java.lang.*;
+import java.io.*;
+
+class Paranthesis{
+        
+        //function to print all valid pairs of parantheses by storing the count of opening and closing brackets
+        static void generate_paranthesis(char str[], int pos, int n, int start, int end){
+            if(end == n){
+                for(int i = 0; i < str.length; i++){
+                    System.out.print(str[i]);           //printing all pairs of parantheses
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:15]: Add comments above the line throughout the code.

File: Code/Java/valid_parantheses.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            if( n > 0)
+                generate_paranthesis(ans, 0, n, 0 , 0);
+            
+            return;
+        }
+    
+    	public static void main (String[] args) throws java.lang.Exception
+    	{
+    	        Scanner sc = new Scanner(System.in); 
+                int n = 0;
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:44]: Remove this comment

File: Code/Java/valid_parantheses.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                n = sc.nextInt();                               //input from the users
+                char [] arr = new char[2*n];                    //generating a character array for storing parantheses
+        		valid_paranthesis(arr, n);                      //calling the function
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:47]: Remove this comment

File: Code/Java/valid_parantheses.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    	}
+}
+
+/*Test Case 1:
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:51]:
```suggestion
/*
Test Cases: 

```
Follow this format
Test Case
1. 
Input : 
Output :

File: Code/Java/valid_parantheses.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                {{}{{}}}
+                {{{}}}{}
+                {{{}}{}}
+                {{{}{}}}
+                {{{{}}}}
+
+               
+Time Complexity: O(2^n)
+
+Space Complexity: O(n)
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:83]: Specify n

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by shahina-bano** [comment]
Done @raksha009 :))

File: Code/Java/valid_parantheses.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    
+    	public static void main (String[] args) throws java.lang.Exception
+    	{
+    	        Scanner sc = new Scanner(System.in); 
+                int n = 0;
+                n = sc.nextInt();        
+                //generating a character array for storing parantheses
+                char [] arr = new char[2*n];                   
+        		valid_paranthesis(arr, n);
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:58]: Indentation!

File: Code/Java/valid_parantheses.java [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+                if(start > end){
+                     //checking the condition for inserting the closing parantheses
+                    str[pos] = '}';
+                    //calling the function for the next position using recursion
+                    generate_paranthesis(str, pos+1, n, start, end+1);     
+                }
+                //checking the condition for inserting the opeing paranthesis
+                if(start < n){                          
+                    str[pos] = '{';
+                     //calling the function for the next position using recursion
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:37]: Remove this comment.

File: Code/Java/valid_parantheses.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+                }
+                
+                System.out.println();
+                return;
+            }
+            else{
+                if(start > end){
+                     //checking the condition for inserting the closing parantheses
+                    str[pos] = '}';
+                    //calling the function for the next position using recursion
```
Comment (raksha009) [inline Code/Java/valid_parantheses.java:31]: Remove this comment

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by raksha009 — APPROVED** [review]
LGTM

**Pull request merged**